catalyst-gateway-crates/c509-certificate/examples/cli/main.rs

```diff
@@ -0,0 +1,350 @@
+//! C509 certificate CLI
+
+use std::{
+    fs::{self, File},
+    io::Write,
+    path::PathBuf,
+    str::FromStr,
+};
+
+use asn1_rs::{oid, Oid};
+use c509_certificate::{
+    c509_big_uint::UnwrappedBigUint,
+    c509_extensions::Extensions,
+    c509_issuer_sig_algo::IssuerSignatureAlgorithm,
+    c509_name::{rdn::RelativeDistinguishedName, Name, NameValue},
+    c509_subject_pub_key_algo::SubjectPubKeyAlgorithm,
+    c509_time::Time,
+    signing::{PrivateKey, PublicKey},
+    tbs_cert::TbsCert,
+};
+use chrono::{DateTime, Utc};
+use clap::{Parser, Subcommand};
+use hex::ToHex;
+use minicbor::Decode;
+use rand::Rng;
+use serde::{Deserialize, Serialize};
+
+/// A struct for CLI.
+#[derive(Parser)]
+#[command(version, about, long_about = None)]
+struct Cli {
+    /// Optional subcommands.
+    #[command(subcommand)]
+    command: Option<Commands>,
```

Comment:
Seems it is redundant to have this `commands` field here.
Let's change this `Cli` struct to `enum` with copying basically all enum options from `Commands` and remove `Commands` type.
```rust
enum Cli {
Generate{...},
Verify{...},
....
}
```